control-plane-operator/controllers/hostedcontrolplane/cco/operator.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/assets"
+	"github.com/openshift/hypershift/support/certs"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/metrics"
+	"github.com/openshift/hypershift/support/util"
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+var (
+	cloudCredentialOperatorMetricsService = assets.MustService(content.ReadFile, "assets/01-service.yaml")
+	cloudCredentialOperatorDeployment     = assets.MustDeployment(content.ReadFile, "assets/03-deployment.yaml")
+)
+
+func cloudCredentialOperatorLabels() map[string]string {
+	return map[string]string{"app": "cloud-credential-operator", hyperv1.ControlPlaneComponent: "cloud-credential-operator"}
+}
+
+func ReconcileCloudCredentialOperatorMetricsService(svc *corev1.Service, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(svc)
+
+	// The service is assigned a cluster IP when it is created.
+	// This field is immutable as shown here: https://github.com/kubernetes/api/blob/62998e98c313b2ca15b1da278aa702bdd7b84cb0/core/v1/types.go#L4114-L4130
+	// As such, to avoid an error when updating the object, only update the fields CCO specifies.
+	cloudCredentialOperatorMetricsServiceDeepCopy := cloudCredentialOperatorMetricsService.DeepCopy()
+	svc.Spec.Ports = cloudCredentialOperatorMetricsServiceDeepCopy.Spec.Ports
+	svc.Spec.Type = cloudCredentialOperatorMetricsServiceDeepCopy.Spec.Type
+	svc.Spec.Selector = cloudCredentialOperatorMetricsServiceDeepCopy.Spec.Selector
+	svc.Spec.SessionAffinity = cloudCredentialOperatorMetricsServiceDeepCopy.Spec.SessionAffinity
+
+	return nil
+}
+
+func ReconcileCloudCredentialOperatorDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef, cloudCredentialOperatorImage, kubeRBACProxyImage, awsPodIdentityWebhookImage, releaseVersion string, dc config.DeploymentConfig, availabilityProberImage string, apiPort *int32) error {
+	ownerRef.ApplyTo(deployment)
+	deployment.Spec = cloudCredentialOperatorDeployment.DeepCopy().Spec
+	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: "kubeconfig",
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  manifests.KASServiceKubeconfigSecret(deployment.Namespace).Name,
+				DefaultMode: pointer.Int32(0640),
+			},
+		},
+	})
+	for i, volume := range deployment.Spec.Template.Spec.Volumes {
+		if volume.Name == "cloud-credential-operator-serving-cert" && volume.Secret != nil {
+			deployment.Spec.Template.Spec.Volumes[i].Secret.SecretName = manifests.MetricsClientCertSecret(deployment.Namespace).Name
```

Comment:
Looks like the `kube-rbac-proxy` guards the metrics endpoint, so I'm assuming we want this cert here?